### DISCUSSION TITLE
information theory functions

### DISCUSSION THREAD

**Discussion by mjrmason** [RfastOfficial/Rfast#69] [Ideas]
Hi, 
I have had a lot of success using Rfast to implement correlation based network constructions methods. I am wondering if there will be any added functions for information based  measures. Entropy measures for single feature is easy enough with Rfast. functions based on combinations of features like MI and conditional MI are harder.  Is there going to be any push to add these or any way to to look at combinations of pairs or triplets in future releases? I am mainly interested in the case where binning has already been done and one is working with a matrix of factors indicating bin but I expect many people might be interested in fast functions for binning as well.

Thanks for an awesome package!

PS if you are interested to see a use case where Rfast was crucial a network construction method you can check out [this repo](https://systems-methods.github.io/icWGCNA/).

**Comment by statlink** [comment]
Hi Mason,

Thanks for your good wors. We have functions for conditional MI, with categorical data only. The MI is proportional to the G^2 test statistic. In fact, we also compute the p-value using permutation, so that is pretty similar to the MI. Is this good enough? 
As for the binning, this is done internally (in C++) in those G^2 functions. 

Michail

**Reply by mjrmason** [reply]
Thanks so much, Michail! 
I didn't realize this! This might work. I don't see a function for a conditional g2tests that works on all pairwise combinations of features with a subset for conditioning on. I do see the g2test_univariate which works for all pairwise combination but is not for conditional GS. Currently I think I would have to apply the conditional g2tests function to all pairs of features. Is that correct 
What do you think?
Cheers,
Mike

**Reply by statlink** [reply]
Hi Mike, the conditional 2test requires a conditioning set of variable(s), by default. No, your question is not correct. 
The g2test is for pairs of variables, but the conditional g2test requires at least 1 conditioning variable. 

Apologies for the late reply.
Michail

**Reply by mjrmason** [reply]
Thanks so much!
